### PR TITLE
feat: Add migrated label to metrics

### PR DESCRIPTION
**Pull request by nearora-msft**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
/kind feature
> /kind flake

**What this PR does / why we need it**:
This PR is an extension to kubernetes-csi/csi-lib-utils#77 to send "migrated" label in the metrics

**Which issue(s) this PR fixes**:
Partially Addressed # kubernetes/kubernetes#98279

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Set the value of "migrated" field in the metrics to true or false to indicate if the call is a migration enabled feature or not
```

### CODE REVIEW

File: cmd/csi-attacher/main.go [hunk 1]
```diff
@@ -151,6 +145,22 @@ func main() {
 	}
 	klog.V(2).Infof("CSI driver name: %q", csiAttacher)
 
+	translator := csitrans.New()
+	if translator.IsMigratedCSIDriverByName(csiAttacher) {
+		metricsManager = metrics.NewCSIMetricsManagerWithOptions("" /* driverName */, metrics.WithMigration())
```
Comment (Jiawei0227) [inline cmd/csi-attacher/main.go:150]: nit: set driver name to `csiAttacher`

**Comment by Jiawei0227** [comment]
/lgtm
/assign @msau42 
I think it looks good to me. Will leave it to @msau42 for approval. Please also make changes if there is any comments that you got from csi-provisioner so the implementation is in sync.

**Comment by Jiawei0227** [comment]
I have tested locally and it worked well.

**Comment by msau42** [comment]
I'll wait until comments are resolved in the external-provisioner PR and applied to the other sidecars.

File: cmd/csi-attacher/main.go (hunk 1 above)
Comment (nearora-msft) [reply cmd/csi-attacher/main.go]: Done

**Comment by nearora-msft** [comment]
> I'll wait until comments are resolved in the external-provisioner PR and applied to the other sidecars.

Applied the changes concerning the comments in the external-provisioner

**Pull request merged**